spec/features/post_auth_spec.rb

```diff
@@ -141,6 +141,17 @@
       end
     end
 
+    shared_examples 'it saves the right logging information' do
+      context 'With a user in the database' do
+        it_behaves_like 'it saves the right logging information inner'
+      end
+
+      context 'Without a user in the database' do
+        let!(:create_user) { nil }
+        it_behaves_like 'it saves the right logging information inner', has_user=false
```

Comment:
I think the de-duplication is premature here - I worry these specs aren't easily readable with these changes.

What do you think about writing these two cases (with a user / without a user) out in full? Did it look too long / confusing?

Reply:
Addressed, shifted to to directed tests.
I wasn't entirely happy with this approach anyway